x/docs/internal/keeper/keeper_test.go

```diff
@@ -451,7 +520,107 @@ func TestKeeper_DocumentsIterator_ExistingList(t *testing.T) {
 // --- Document receipts
 // ----------------------------------
 
-func TestKeeper_SaveDocumentReceipt_EmptyList(t *testing.T) {
+func TestKeeper_SaveDocument(t *testing.T) {
+	tests := []struct {
+		name          string
+		empty         bool
+		document      types.Document
+		receipt       types.DocumentReceipt
+		differentUuid bool
+		newReceipt    types.DocumentReceipt
+		//wantErr bool
+	}{
+		{
+			"empty list",
+			true,
+			TestingDocument,
+			TestingDocumentReceipt,
+			false,
+			types.DocumentReceipt{},
+		},
+		{
+			"existing receipt",
+			false,
+			types.Document{},
+			TestingDocumentReceipt,
+			false,
+			types.DocumentReceipt{},
+		},
+		{
+			"existing Receipt different Uuid",
+			false,
+			TestingDocument,
+			TestingDocumentReceipt,
+			true,
+			types.DocumentReceipt{
+				UUID:         TestingDocumentReceipt.UUID + "-new",
+				Sender:       TestingDocumentReceipt.Sender,
+				Recipient:    TestingDocumentReceipt.Recipient,
+				TxHash:       TestingDocumentReceipt.TxHash,
+				DocumentUUID: TestingDocument.UUID,
+				Proof:        TestingDocumentReceipt.Proof,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cdc, ctx, k := SetupTestInput()
+
+			if tt.empty {
+				store := ctx.KVStore(k.StoreKey)
+
+				require.NoError(t, k.SaveDocument(ctx, tt.document))
```

Comment:
If you must SaveDocument in multiple tests, try getting this line outside of various if/else cases: the less redundant the better.